src/internal/user_api/mod.rs

```diff
@@ -56,7 +56,7 @@ impl TryFrom<u64> for DeviceId {
         //Validate the range of the device ID to always be positive, but also be
         //less than i64 (i.e. no high bit set) for compatibility with other
         //languages (i.e. Java)
-        if device_id < 1 || device_id > (std::i64::MAX as u64) {
+        if device_id < 1 || device_id > (i64::max_value() as u64) {
```

Comment:
This surprises me a bit, but I'm guessing that `max_value()` is a const fn so there is no difference at runtime. Has `MAX` been deprecated, or why is this better?

Reply:
Quick googling suggests that they're equivalent, so I don't think either is better. Not sure why there's a lint for it, but it shouldn't hurt? I can change this one back if you prefer.

Reply:
No. No problem, I was just wondering. This is fine by me.